packages/js/components/src/date-time-picker/date-time-picker.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { createElement, useState, useEffect } from '@wordpress/element';
+import {
+	Dropdown,
+	DateTimePicker as WpDateTimePicker,
+} from '@wordpress/components';
+import moment from 'moment';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { default as DateInput } from '../calendar/input';
+
+export type DateTimeProps = {
+	onChange: ( date: string ) => void;
+	dateTimeFormat?: string;
+	disabled?: boolean;
+	currentDate?: string;
+	is12Hour?: boolean;
+} & React.HTMLAttributes< HTMLDivElement >;
+
+export const DateTimePicker: React.FC< DateTimeProps > = ( {
+	onChange,
+	is12Hour = true,
+	dateTimeFormat = is12Hour ? 'MM/DD/YYYY h:mm a' : 'MM/DD/YYYY H:MM',
+	disabled = false,
+	currentDate = new Date().toISOString(),
```

Comment:
I think we should also add a `useEffect` for the `currentDate` prop and update the `dateTime` state when the `currentDate` updates.
So we can update it from the outside of the component as well and not just on initial render.